src/domain/covenant-identity.ts

```diff
@@ -0,0 +1,429 @@
+import {
+  Identity,
+  crypto,
+  address,
+  Psbt,
+  restorerFromEsplora,
+  IdentityType,
+  restorerFromState,
+  Transaction,
+  bip341,
+  toXpub,
+  witnessStackToScriptWitness,
+  checkIdentityType,
+  checkMnemonic,
+  networks,
+  fromXpub,
+} from 'ldk';
+import type {
+  AddressInterface,
+  IdentityInterface,
+  IdentityOpts,
+  Restorer,
+  EsploraRestorerOpts,
+  NetworkString,
+  StateRestorerOpts,
+  Mnemonic,
+} from 'ldk';
+import type { BlindingDataLike } from 'liquidjs-lib/src/psbt';
+import { evaluate } from '../descriptors';
+import type { Result } from '../descriptors/ast';
+import type { Context } from '../descriptors/preprocessing';
+import { SLIP77Factory } from 'slip77';
+import * as ecc from 'tiny-secp256k1';
+import type { BIP32Interface } from 'bip32';
+import BIP32Factory from 'bip32';
+import { mnemonicToSeedSync } from 'bip39';
+import type { ScriptInputsNeeds } from './script-analyser';
+import { analyzeTapscriptTree } from './script-analyser';
+
+// slip13: https://github.com/satoshilabs/slips/blob/master/slip-0013.md#hd-structure
+function namespaceToDerivationPath(namespace: string): string {
+  const hash = crypto.sha256(Buffer.from(namespace));
```

Comment:
We should also fingerprint the template itself imho 

Sort of versioning of the contract/template changes overtime